mrp_production_operation_injection/models/mrp_production.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    display_inject_workorder = fields.Boolean(
+        compute="_compute_display_inject_workorder"
+    )
+
+    @api.depends("state", "bom_id", "bom_id.operation_ids", "workorder_ids")
+    def _compute_display_inject_workorder(self):
+        for production in self:
+            production.display_inject_workorder = (
+                production.state in ["confirmed", "progress", "to_close"]
+                and production.bom_id.operation_ids
+                and production.workorder_ids
+            )
+
+    def action_open_workorder_injector(self):
+        self.ensure_one()
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "mrp_production_operation_injection.mrp_workorder_injector_action"
+        )
+        ctx = self.env.context.copy()
+        ctx["default_production_id"] = self.id
+        action.update({"context": ctx})
+        return action
+
+    def _prepare_injected_workorder_values(self, operation):
+        self.ensure_one()
+        return {
+            "name": operation.name,
+            "production_id": self.id,
+            "workcenter_id": operation.workcenter_id.id,
+            "product_uom_id": self.product_uom_id.id,
+            "operation_id": operation.id,
+            "state": "pending",
+        }
+
+    def _add_workorder(self, operation, previous_workorder):
+        self.ensure_one()
+        following_workorders = self.workorder_ids.filtered(
+            lambda w: w.sequence > previous_workorder.sequence
+        )
+        next_workorder = fields.first(following_workorders)
+        # Prepare creation of new workorder
+        workorder_values = self._prepare_injected_workorder_values(operation)
+        workorder_values["sequence"] = previous_workorder.sequence + 1
+        workorder_values["next_work_order_id"] = next_workorder.id
+        # FIXME: state computation is not good in Odoo anyway so handle
+        #  only most 'probable' cases only
+        if next_workorder.state in ["ready", "progress"]:
+            workorder_values["state"] = "ready"
+        # Update following workorders sequence before create to make sure workorders
+        #  can be ordered properly for _action_confirm (cf override in mrp_workorder)
+        for wo in following_workorders:
+            wo.sequence += 1
```

Comment:
can it happen that previous_workorder.sequence == next_workorder.sequence?
In such case after this, you would have new_workorder.sequence == next_workorder.sequence.
Is that ok?
Otherwise smth like that maybe?
```python
next_sequence = previous_workorder.sequence + 1
next_sequences = range(next_sequence, next_sequence + len(following_workorders))
for wo, wo_sequence in zip(following_workorders, next_sequences):
    wo.sequence = wo_sequence
```

Reply:
I would say we don't care because this case would have to be handled by the module providing workorder sequence: https://github.com/OCA/manufacture/tree/15.0/mrp_workorder_sequence